Get it to build

stippi opened the pull request:
These are all changes necessary to get it building. Please have a close look. My suspicion is that you haven't quite gotten the hang of .cpp and .h files and where you need which. As for the Makefile changes, it is probably useful to understand what I did there and why, but integrating the changes with Haiku later, it will be Jamfiles anyway. It's a bit different, but the Jamfiles are a complex build system which handles all of the various ways to build Haiku. Most of the Haiku devs themselves are intimidated by the build system, elegant as it may be, once you know it deeply.

The next task is getting the app to run. :-)

deepango committed:
Try to integrate harfbuzz

deepango committed:
Commented clock_t

deepango committed:
Made changes Bitmap

deepango committed:
Fixed errors

digib0y commented:
You created this pull for my master branch, is there any possible way I can shift this PR to my broken branch?

stippi commented:
Why can't you just switch to the master branch, too?

I need to pick up my kid from a friend's house, will be back soon. Hoping we can spend some time in IRC later.

stippi committed:
Removed generated files

stippi committed:
Remove alternative build system

stippi committed:
Added .gitignore which ignores generated files

stippi committed:
Merge branch 'master' into broken

stippi committed:
Removed generated files from tree

stippi committed:
Removed unneeded headers from MainWindow.cpp

Since the code in hbshaper.h then ends up in MainWindow.o and
MainView.o, the linker will complain that there are multible
clashing symbols and refuse to link. So hbshaper.h as is can only
be included from one source file at the moment. To fix this, it
would have to be devided into an .h with just the class declaration
and a .cpp with the implementation. Like the other files.

stippi committed:
Fixed the Makefile

Fixed various declarations in the Makefile. This gets the demo app
linked, finally.

deepango committed:
New changes to fix the bug
